src/onion/shortcuts.c

```diff
@@ -224,7 +224,7 @@ onion_connection_status onion_shortcut_response_file(const char *filename,
       if (*end)
         ends = atol(end);
       else
-        ends = length;
+        ends = length > 1 ? length - 1 : 0;
```

Comment:
May be it will make sense to report `416 Requested Range Not Satisfiable` in the case of zero length